File: Easy/455_Assign_Cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        
        m = len(g)
        n = len(s)

        g.sort()
        s.sort()

        i = 0
        j = 0
        ans = 0

        while i < m and j < n:
            if g[i] <= s[j]:
                ans += 1
                i += 1
                j += 1
            else:
                j += 1
        return ans

File: Easy/455_Assign_Cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        
        int m = g.size();
        int n = s.size();

        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int i = 0;
        int j = 0;
        int ans = 0;

        while(i < m && j < n)
        {
            if(g[i] <= s[j])
            {
                ans++;
                i++;
                j++;
            }else if(g[i] > s[j])
            {
                j++;
            }
        }
        return ans;
    }
};
